src/dotnet/commands/dotnet-update/tool/UpdateToolCommand.cs

```diff
@@ -0,0 +1,246 @@
+
+            DirectoryPath? toolPath = null;
+            if (_toolPath != null)
+            {
+                toolPath = new DirectoryPath(_toolPath);
+            }
+
+            (IToolPackageStore toolPackageStore, IToolPackageInstaller toolPackageInstaller) =
+                _createToolPackageStoreAndInstaller(toolPath);
+            IShellShimRepository shellShimRepository = _createShellShimRepository(toolPath);
+
+
+            IToolPackage oldPackage;
+            try
+            {
+                oldPackage = toolPackageStore.EnumeratePackageVersions(_packageId).SingleOrDefault();
+                if (oldPackage == null)
+                {
+                    throw new GracefulException(
+                        messages: new[]
+                        {
+                            string.Format(
+                                LocalizableStrings.ToolNotInstalled,
+                                _packageId),
+                        },
+                        isUserError: false);
+                }
+            }
+            catch (InvalidOperationException)
+            {
+                throw new GracefulException(
+                    messages: new[]
+                    {
+                        string.Format(
+                            LocalizableStrings.ToolHasMultipleVersionsInstalled,
+                            _packageId),
+                    },
+                    isUserError: false);
+            }
+
+            FilePath? configFile = null;
+            if (_configFilePath != null)
+            {
+                configFile = new FilePath(_configFilePath);
+            }
+
+            using (var scope = new TransactionScope(
+                TransactionScopeOption.Required,
+                TimeSpan.Zero))
+            {
+                RunWithHandlingUninstallError(() =>
+                {
+                    foreach (CommandSettings command in oldPackage.Commands)
+                    {
+                        shellShimRepository.RemoveShim(command.Name);
+                    }
+
+                    oldPackage.Uninstall();
+                });
+
+                RunWithHandlingInstallError(() =>
+                {
+                    IToolPackage newInstalledPackage = toolPackageInstaller.InstallPackage(
+                        packageId: _packageId,
+                        targetFramework: _framework,
+                        nugetConfig: configFile,
+                        source: _source,
+                        verbosity: _verbosity);
+
+                    foreach (CommandSettings command in newInstalledPackage.Commands)
+                    {
+                        shellShimRepository.CreateShim(command.Executable, command.Name);
+                    }
+
+                    PrintSuccessMessage(oldPackage, newInstalledPackage);
+                });
+
+                scope.Complete();
+            }
+
+            return 0;
+        }
+
+        private void ValidateArgument()
```

Comment:
nit: `ValidateArguments` because it validates more than a singular argument.

Reply:
Changed as suggested